src/main/java/net/sf/jabref/logic/formatter/bibtexfields/DateFormatter.java

```diff
@@ -0,0 +1,64 @@
+package net.sf.jabref.logic.formatter.bibtexfields;
+
+import net.sf.jabref.logic.formatter.Formatter;
+
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.time.temporal.TemporalAccessor;
+
+/**
+ * This class transforms date to the format yyyy-mm-dd or yyyy-mm..
+ */
+public class DateFormatter implements Formatter {
+    @Override
+    public String getName() {
+        return "Date";
+    }
+
+    /**
+     * Format date string to yyyy-mm-dd or yyyy-mm. Keeps the existing String if it does not match one of the following
+     * formats:
+     *  //@formatter:off
```

Comment:
Do we need to use these tags even inside Javadoc? For me, then the formatting is fundamentally flawed...